lib/OptimizationSciPy/src/OptimizationSciPy.jl

```diff
@@ -0,0 +1,1381 @@
+module OptimizationSciPy
+
+using Reexport
+@reexport using Optimization
+using Optimization.SciMLBase
+using PythonCall
+
+const scipy = PythonCall.pynew()
+
+function __init__()
+    PythonCall.pycopy!(scipy, pyimport("scipy"))
+end
+
+function ensure_julia_array(x, ::Type{T}=Float64) where T
+    x isa Vector{T} && return x
+    return convert(Vector{T}, x isa Py ? pyconvert(Vector, x) : x)
+end
+
+function safe_get_message(result)
+    try
+        pyconvert(String, result.message)
+    catch
+        "Optimization completed"
+    end
+end
+
+function safe_to_float(x)
+    x isa Float64 && return x
+    try
+        return pyconvert(Float64, x)
+    catch
+        
+        if pyhasattr(x, "item")
+            try
+                return pyconvert(Float64, x.item())
+            catch
+            end
+        end
+        try
+            return pyconvert(Float64, pybuiltins.float(x))
+        catch
+            error("Cannot convert Python object to Float64: $(typeof(x))")
+        end
+    end
+end
+
+function extract_stats(result, time_elapsed)
+    stats_dict = Dict{Symbol, Any}(
+        :iterations => 0,
+        :time => time_elapsed,
+        :fevals => 0,
+        :gevals => 0,
+        :hevals => 0
+    )
+    if pyhasattr(result, "nit")
+        try
+            stats_dict[:iterations] = pyconvert(Int, result.nit)
```

Comment:
why would this need a try catch? It seems if you already check for the existence this is fine without it